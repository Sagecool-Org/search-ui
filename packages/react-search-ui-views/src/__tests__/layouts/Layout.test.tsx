import React from "react";
import { render } from "@testing-library/react";
import { Layout } from "../..";

it("renders correctly", () => {
  const { container } = render(
    <Layout
      header={<div>Header</div>}
      sideContent={
        <div>
          <div>Side Content</div>
        </div>
      }
      bodyContent={<div>Body Content</div>}
      bodyHeader={<div>Body Header</div>}
      bodyFooter={<div>Body Footer</div>}
    />
  );
  expect(container).toMatchSnapshot();
});

it("will accept children instead of bodyContent", () => {
  const { container } = render(
    <Layout
      header={<div>Header</div>}
      sideContent={
        <div>
          <div>Side Content</div>
        </div>
      }
      bodyHeader={<div>Body Header</div>}
      bodyFooter={<div>Body Footer</div>}
    >
      <div>Body Content</div>
    </Layout>
  );
  expect(container).toMatchSnapshot();
});

it("renders with className prop applied", () => {
  const { container } = render(<Layout className="test-class" />);
  expect(container.firstChild).toHaveClass("test-class");
});
