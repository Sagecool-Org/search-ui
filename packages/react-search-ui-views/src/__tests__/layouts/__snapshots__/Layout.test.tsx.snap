// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="sui-layout"
  >
    <div
      class="sui-layout-header"
    >
      <div
        class="sui-layout-header__inner"
      >
        <div>
          Header
        </div>
      </div>
    </div>
    <div
      class="sui-layout-body"
    >
      <div
        class="sui-layout-body__inner"
      >
        <button
          class="sui-layout-sidebar-toggle"
          hidden=""
          type="button"
        >
          Show Filters
        </button>
        <div
          class="sui-layout-sidebar"
        >
          <button
            class="sui-layout-sidebar-toggle"
            hidden=""
            type="button"
          >
            Save Filters
          </button>
          <div>
            <div>
              Side Content
            </div>
          </div>
        </div>
        <div
          class="sui-layout-main"
        >
          <div
            class="sui-layout-main-header"
          >
            <div
              class="sui-layout-main-header__inner"
            >
              <div>
                Body Header
              </div>
            </div>
          </div>
          <div
            class="sui-layout-main-body"
          >
            <div>
              Body Content
            </div>
          </div>
          <div
            class="sui-layout-main-footer"
          >
            <div>
              Body Footer
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`will accept children instead of bodyContent 1`] = `
<div>
  <div
    class="sui-layout"
  >
    <div
      class="sui-layout-header"
    >
      <div
        class="sui-layout-header__inner"
      >
        <div>
          Header
        </div>
      </div>
    </div>
    <div
      class="sui-layout-body"
    >
      <div
        class="sui-layout-body__inner"
      >
        <button
          class="sui-layout-sidebar-toggle"
          hidden=""
          type="button"
        >
          Show Filters
        </button>
        <div
          class="sui-layout-sidebar"
        >
          <button
            class="sui-layout-sidebar-toggle"
            hidden=""
            type="button"
          >
            Save Filters
          </button>
          <div>
            <div>
              Side Content
            </div>
          </div>
        </div>
        <div
          class="sui-layout-main"
        >
          <div
            class="sui-layout-main-header"
          >
            <div
              class="sui-layout-main-header__inner"
            >
              <div>
                Body Header
              </div>
            </div>
          </div>
          <div
            class="sui-layout-main-body"
          >
            <div>
              Body Content
            </div>
          </div>
          <div
            class="sui-layout-main-footer"
          >
            <div>
              Body Footer
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
